src/utilities.py

```diff
@@ -199,49 +197,47 @@ def get_filename(file: Gio.File):
 
 
 def optimize_limits(self):
+    figure_settings = self.get_figure_settings()
     axes = [
-        [direction, False, []]
+        [direction, False, [], [],
+         figure_settings.get_property(f"{direction}_scale")]
         for direction in ["bottom", "left", "top", "right"]
     ]
     for item in self.get_data():
+        if item.props.item_type != "Item":
+            continue
         for index in item.xposition * 2, 1 + item.yposition * 2:
             axes[index][1] = True
-            axes[index][2].append(item)
-
-    for count, (direction, used, items) in enumerate(axes):
+            data = numpy.asarray(item.ydata if index % 2 else item.xdata)
+            nonzero_data = list(filter(lambda x: (x != 0), data))
+            axes[index][2].append(
+                nonzero_data[numpy.isfinite(nonzero_data)].min()
+                if axes[index][4] in (1, 4) and len(nonzero_data) > 0
+                else data[numpy.isfinite(data)].min(),
+            )
```

Comment:
I forgot to convert non_zero data to a numpy array, meaning it will error out on line 214 if conditions of line 215 are met.

Probably better to filter out the inf values centrally first on data, instead of on data and nonzero data infinitely. We don't want to use that for finding the limits anyway.

```suggestion
            data = numpy.asarray(item.ydata if index % 2 else item.xdata)
            data = data[numpy.isfinite(data)]
            nonzero_data = numpy.array([value for value in data if value != 0])
            axes[index][2].append(
                min(nonzero_data)
                if axes[index][4] in (1, 4) and len(nonzero_data) > 0
                else min(data)
            )
```